tests/saitests/py3/sai_qos_tests.py

```diff
@@ -585,7 +590,7 @@ def fill_leakout_plus_one(
     if asic_type in ['cisco-8000']:
         queue_counters_base = sai_thrift_read_queue_occupancy(
             test_case.dst_client, "dst", dst_port_id)
-        max_packets = 500
+        max_packets = 2000
```

Comment:
can integrate this Cisco revision leakout utils fucntion with "def dynamically_compensate_leakout()"?
The latter first sends the desired test traffic, then checks "TRANSMITTED_PKTS" to see how many packets were leaked, then compensates all packets at once. It can save a little bit test time compare with fill_leakout_plus_one.

Or can explain any benefit for cisco revision one to fill_leakout_plus_one first , then send desired test traffic?

Reply:
Transmitted packets can increment due to control traffic as well. In the case of the PCBBPFCTest at least, this effect is worsened by the ICMP heartbeats for dualtor. 

Some tests also require a single packet filled in the queue anyway to test low-value Buffer pool or queue watermarks. The dynamically_compensate_leakout technique devolves to the same as fill_leakout_plus_one in this scenario. So the max retries for cisco-8122 would still be coded similarly as 2000 for those tests. 

I can consider such a major leakout-revision in another PR if needed, as that'd require a lot of tuning on other devices. I'd also likely need to bump up the margins on most of the tests to account for the background traffic.